server/ocsp.go

```diff
@@ -0,0 +1,454 @@
+// Copyright 2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"golang.org/x/crypto/ocsp"
+)
+
+const (
+	DefaultOCSPCheckInterval = 24 * time.Hour
```

Comment:
Late to comment on that, but we need to think harder when exporting this (applies to all other OCSP things here). I think you had to export because you decided to have the tests in the `/test` package, but I am a big proponent of having tests under `/server`, along with the file that is being tested. It also gives you the flexibility to access server internals if needed.

Once things are exported, we can't fix/change them easily without having the risk to break users apps, which we have to realize that a lot of users actually embed the NATS server.

Reply:
+1 these can be made private